src/main/java/com/marklogic/spark/reader/MarkLogicPartitionReader.java

```diff
@@ -142,4 +149,14 @@ private JacksonParser newJacksonParser(StructType schema) {
         final Seq<Filter> filters = JavaConverters.asScalaIterator(new ArrayList().iterator()).toSeq();
         return new JacksonParser(schema, jsonOptions, allowArraysAsStructs, filters);
     }
+
+    private RowManager selectRowManagerFromDataHosts() {
```

Comment:
This is the logic for choosing a host to use. I've verified it works against a multi-node cluster, and I'm going to provide test support for that in a follow-up PR that puts a docker-compose file in place for building a 3-node cluster.